test/counts.jl

```diff
@@ -102,14 +102,24 @@ StatsBase.addcounts_radixsort!(cm,xx2)
 
 # testing the Dict-based addcounts
 cm = Dict{Int, Int}()
+cm_itr = Dict{Int, Int}()
 StatsBase.addcounts_dict!(cm,xx)
-@test cm == Dict(1 => 200_000, 3 => 100_000, 6 => 100_000)
+StatsBase.addcounts_dict!(cm_itr,skipmissing(xx))
+@test cm_itr == cm == Dict(1 => 200_000, 3 => 100_000, 6 => 100_000)
 
+# test countmap for general iterators
 cm = countmap(x, weights(w))
 @test cm["a"] == 5.5
 @test cm["b"] == 4.5
 @test cm["c"] == 3.5
 
+xx_missing = skipmissing([missing, "b", "a", "a", "b", "c"])
+cm_missing = countmap(xx_missing)
+
+@test cm_missing["a"] == 2
+@test cm_missing["b"] == 2
+@test cm_missing["c"] == 1
```

Comment:
Sorry, two last things with tests:
- This should be put above, with the equivalent test for vectors. Here it's in the middle of tests for the weighted variant.

- Also, could you check that `cm_missing isa Dict{String, Int}`, and that something like `countmap((i for i in x)` gives a `Dict{Any,Int}` instead? That's maybe not ideal (`String` would be better in both cases), but in terms of implementation we would have to call `collect` on the iterator first to find a more specific type when the iterator doesn't have one. Anyway, better test this kind of detail as end users can end up relying on it and we shouldn't change it later without noticing.